app/components/safe_area_view/safe_area_view.ios.js

```diff
@@ -57,23 +58,52 @@ export default class SafeAreaIos extends PureComponent {
     componentDidMount() {
         this.mounted = true;
 
-        SafeArea.addEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        Dimensions.addEventListener('change', this.getSafeAreaInsets);
         EventEmitter.on('update_safe_area_view', this.getSafeAreaInsets);
+        if (EphemeralStore.safeAreaInsets.portait === null || EphemeralStore.safeAreaInsets.landscape === null) {
+            SafeArea.addEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        }
+
         this.keyboardDidShowListener = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
         this.keyboardDidHideListener = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
 
         this.getSafeAreaInsets();
     }
 
     componentWillUnmount() {
-        SafeArea.removeEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        Dimensions.removeEventListener('change', this.getSafeAreaInsets);
         EventEmitter.off('update_safe_area_view', this.getSafeAreaInsets);
+        SafeArea.removeEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
```

Comment:
is there any problem if you try to remove an eventlistener that isn't there? most likely not, but i'd double check

Reply:
nope, this is fine